acceptable/_service.py

```diff
@@ -2,30 +2,88 @@
 # GNU Lesser General Public License version 3 (see the file LICENSE).
 
 """acceptable - Programatic API Metadata for Flask apps."""
+from acceptable import _validation
+
+
+class APIMetadata:
+    """Global datastructure for all services.
+
+    Provides a single point to register apis against, so we can easily inspect
+    and verify uniqueness.
+    """
+
+    def __init__(self):
+        self.services = {}
+        self.api_names = set()
+        self.urls = set()
+
+    def register_service(self, name, group):
+        if (name, group) not in self.services:
+            self.services[name, group] = {}
+
+    def register_api(self, name, group, api):
+        assert api.name not in self.api_names, (
```

Comment:
I know this is moved from elsewhere, but maybe `if not foo: raise AssertionError(...)` instead (here and elsewhere), since `assert` may be compiled out.

Reply:
Hmm, I wonder if having a custom exception is what we want here. Assertions were used previously, I just carried on their usage. As this is all import time code, then it kinda made sense, but  I think we should add a new Exception:

    class Unacceptable(Exception):
        pass